tests/Rules/Deprecations/TypeHintDeprecatedInClassMethodSignatureRuleTest.php

```diff
@@ -0,0 +1,29 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Deprecations;
+
+class TypeHintDeprecatedInClassMethodSignatureRuleTest extends \PHPStan\Testing\RuleTestCase
+{
+
+	protected function getRule(): \PHPStan\Rules\Rule
+	{
+		$broker = $this->createBroker();
+
+		return new TypeHintDeprecatedInClassMethodSignatureRule($broker);
+	}
+
+	public function test(): void
+	{
+		require_once __DIR__ . '/data/typehint-deprecated-class-definition.php';
+		$this->analyse(
+			[__DIR__ . '/data/typehint-deprecated-class.php'],
+			[
+				['Usage of deprecated class TypeHintDeprecatedInFunctionSignature\DeprecatedProperty in argument #0.', 13],
```

Comment:
Please use the same message pattern as in PHPStan:

```
| 5    | Parameter $foo of method HelloWorld::sayHello() has invalid typehint type Foo. |
| 5    | Return typehint of method HelloWorld::sayHello() has invalid type Foo.         |
```

Something like:

* `Parameter $foo of method HelloWorld::sayHello() uses deprecated class %s as its typehint.`
* `Return typehint of method HelloWorld::sayHello() uses deprecated class %s as its typehint.`

Also use `interface` vs `class` correctly. Thanks.